mem.go

```diff
@@ -168,21 +168,27 @@ func (m *Message) NumSegments() int64 {
 // Segment returns the segment with the given ID.
 func (m *Message) Segment(id SegmentID) (*Segment, error) {
 	m.mu.Lock()
```

Comment:
While we're here, does it make any performance difference to use a RWMutex and have a fast-path where we don't consult the arena and acquire a RLock?